internal/pkg/api/server/buffer/service/service.go

```diff
@@ -145,8 +145,24 @@ func (*service) ListReceivers(d dependencies.ForProjectRequest, _ *buffer.ListRe
 	return &buffer.ListReceiversResult{Receivers: receivers}, nil
 }
 
-func (*service) DeleteReceiver(dependencies.ForProjectRequest, *buffer.DeleteReceiverPayload) (res *buffer.Receiver, err error) {
-	return nil, &NotImplementedError{}
+func (*service) DeleteReceiver(d dependencies.ForProjectRequest, payload *buffer.DeleteReceiverPayload) (err error) {
+	ctx, store := d.RequestCtx(), d.ConfigStore()
+
+	projectID, receiverID := d.ProjectID(), payload.ReceiverID
+
+	deleted, err := store.DeleteReceiver(ctx, projectID, receiverID)
```

Comment:
Prislo by mi lepsie vratit iba chybu a tu dat podmienku podla typu chyby, tak sa to myslim riesi standardne.

Ked to budes chciet rozsirit, napr. ze `read-only` rola nemoze mazat, a mas vratit `403`, tak to budes zbytocne prerabat.

Reply:
Najlepsi priklad bude standardna kniznica:
- https://goplay.tools/snippet/WCVoIQlwmkF

> abys ten error bez rozmýšlení propagoval dál 

Kludne to mozes bez rozmyslu propagovat dal, to je spravny pristup.

A ak na danu situaciu potrebujes reagovat specialne, ako v nasom pripade, ze nechceme HTTP kod `500`, ale `404`, tak na to reagujes

Reply:
Alebo este inak:
- mame operaciu `delete`
- ak sa podari, tak `err == nil`
- ak sa nepodari, tak je tam nejaky `err`

Iny pripad su funkcie, ktore vratia `result, bool`, napr. nejaky `Find`.
- ak je `bool = true`, nasli sme to
- ak je `bool = false`, nenasli sme to
- pouziva sa to pri operaciach, ktore nemozu skoncit chybou (napr. hladas nieco v nejakom poli, bud to najdes alebo nie)

A tieto 2 pristupy asi nikdy nie je potrebne miesat.

Reply:
Upravil jsem to, ale pořád mi to nepřijde lepší. Můžeme to tak ale nechat, když už ten pattern chceme používat.

Ty utilities v `errors` mi pořád přijdou jako footgun, jsou to rozbité funkce. Chápu že výmluva je performance, ale pokud chceme performance, proč tam vůbec přidáváme tu dynamickou kontrolu typu? Vytváříme value, pak ji castneme na interface (heap allocation) a nakonec používáme runtime type information, abysme zkontrolovali typ (branch + cache miss). Error codes by byly mnohem víc performant, ale pak ztrácíš typovou informaci. Beru to jako limitaci jazyka, protože nemá tagged unions, který by tohle krásně řešily (statically-typed errors, forced error checking, error exhaustivity).

Reply:
ok :slightly_smiling_face: to sme uz diskutovali, nemam ti viac co k tomu povedat.
Vdaka, ze ten `delete` das prec.